ravenframework/JobHandler.py

```diff
@@ -585,8 +620,9 @@ def isFinished(self):
       # that is not done.
       for run in self.__running+self.__clientRunning:
         if run:
-          return False
-
+          if run.identifier.startswith(jobIdentifier) \
```

Comment:
Since the default jobIdentifier='', if it is not provided, this line always True. Please check it.
![image](https://user-images.githubusercontent.com/7321071/219492042-573c4b39-a786-45d1-92d6-cdab206a200a.png)

Reply:
Yes this is right.
See 
```
 @ In, jobIdentifier, string, optional, if specified, only check
         finished runs that start with this text. If not specified check all.
```
If you think it is too tricky I can set to None and then add an ```if``` statement

Reply:
I see your point. But I think the 'if' check here is not very clear to me. For example, if both jobIdentifier and uniqueHandler are provided as input to this function, the run only match unqueHandler and the jobIdentifier is different, it will still return False, it seems to me this is not the intent for this check, right? In addition, I prefer use None as default for jobIdentifier. @alfoa

Reply:
addressed